File: leetcode-submissions/2053_Kth_Distinct_String_in_an_Array.ts
function kthDistinct(arr: string[], k: number): string {
    const mp=new Map<string,number>();
    for(let i=0;i<arr.length;i++){
        if(!mp.has(arr[i])){
            mp.set(arr[i],i);
        }else{
            mp.set(arr[i],mp.get(arr[i])+1);
        }
    }
    let cnt=0;
    for(let i=0;i<arr.length;i++){
        if(mp.get(arr[i])===1){
            cnt++;
        }
        if(cnt==k){
            return arr[i];
        }
    }
return "";
};

File: leetcode-submissions/2053_Kth_Distinct_String_in_an_Array.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        map<string,int>mp;
        for(int i=0;i<arr.size();i++){
            mp[arr[i]]++;

        }
        int cnt=0;

        for(string str:arr){
            if(mp[str]==1){
                cnt++;
            }
            if(cnt==k){
                return str;
            }
        }

       
        
        return "";
    }
};